### DISCUSSION TITLE
[keyball44]ledの彩度を下げると片側がフリーズする

### DISCUSSION THREAD

**Discussion by jwzxr** [Yowkees/keyball#446] [Q&A]
お世話になります。

右側のpromicroに接続し、標準でlayer3の左側に設定されている彩度設定を下げ続けると、途中で左側がフリーズし入力を受け付けなくなります。

左側のpromicroに接続すると左右とも問題なく動作します。

右側に接続し、Remap上でイルミネーションパターンを操作していると復帰します。usb\ttrsケーブルの抜き差しでは復帰しません。リセットボタンも反応しません。

追加で検証しました。
・ファームウェア更新でも改善しませんでした。
・RGB HUEを別のキーマップに配置して実行してと同様の問題が発生します。
・Remap上のLightningでHueを直接入力すると同様の問題が発生します。
・LightningのEffect ModeをSolid Modeにしておくと10程度で復帰します。

当該機能を使用しなければ使用に問題ない為無視しようかとも思いましたが、若干の不安もある為対応したいと考えております。
このような状況の場合、どのような対応策が考えられますか？

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

症状から想像するに、フルカラーLEDの半田付けもしくは、部品自体が故障して電流を引き込んでいるものがあるように感じます。
Remapで公開しているKeyballのTESTファームウェアを書き込むと、赤→緑→青の順で点灯します。このテストで左手側で発光のおかしいLEDはないでしょうか？

お手数ですがご確認お願いいたします。

**Comment by jwzxr** [comment]
ご対応ありがとうございます。

素人目ではありますが、一通り確認しましたところ発光のおかしいLEDは見当たらないように思います。
念のため、画像を添付させていただきます。
![image0](https://github.com/Yowkees/keyball/assets/153560360/db2328b0-ef0f-4551-acbb-65a67197c3a8)
![image1](https://github.com/Yowkees/keyball/assets/153560360/388addf4-e67b-4616-babf-9b18c5dbf4c4)
![image2](https://github.com/Yowkees/keyball/assets/153560360/6a33a6c6-ed57-49a8-8475-aecfcb5955ae)
![image3](https://github.com/Yowkees/keyball/assets/153560360/e546dec1-3e7f-4f9b-93ab-67334f8eba80)
![image4](https://github.com/Yowkees/keyball/assets/153560360/5cdbf4e6-60e7-4e70-a212-cd2cf4e545fd)
![image5](https://github.com/Yowkees/keyball/assets/153560360/48857bc9-73e3-4c0f-9430-334fb2f39e59)

**Comment by jwzxr** [comment]
色々と検証をした結果、フリーズの要因が彩度ではなく明度に依存することがわかりました。

まず、左側のみで発生すると思われていたフリーズ状態ですが、右側でも彩度を０にして明度を１００にするとフリーズが発生します。
逆に明度を８０制限すると彩度を０にしてもフリーズは発生しません。

ファームウェアの最大輝度設定を変更するという以外に対応策がありますでしょうか？

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ProMicroの電気仕様で、1端子あたりの最大電流は1Aだったと記憶しています。
Keyball44には取り付けられる限りのLEDを配置していまして、あまり白っぽい色に設定するとこの最大値に達します。
申し訳ありませんが、LEDにちらつきが出たりフリーズするようでしたら輝度や色を抑え、工夫してご使用いただければ幸いです。

以上、よろしくお願いいたします。
